api/auth.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+message AuthAction {
+  // Should we do white-list or black-list style access control.
+  enum ActionType {
+    // Request matches all rules are allowed, otherwise denied.
+    ALLOW = 0;
+    // Request matches all rules or missing required auth fields are denied,
+    // otherwise allowed.
+    DENY = 1;
+  }
+
+  ActionType action_type = 1;
+
+  // Element type of AndRule/OrRule, it chooses among different type of rule.
+  message Rule {
+    oneof rule_oneof {
```

Comment:
Nit: prefer `rule_specifier` for consistency with other `oneof` naming, e.g. https://github.com/envoyproxy/data-plane-api/blob/master/api/filter/accesslog.proto#L68